### PR TITLE
Build: add names to Travis jobs

### PR DESCRIPTION
**Pull request by gkrizek**
This adds the `name` field to all the TravisCI jobs. This will make it more obvious in the Travis UI what job is failing or passing.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -52,6 +53,7 @@ jobs:
         - set -o errexit; source .travis/lint_06_script.sh
 # ARM
```
Comment (scravy) [inline .travis.yml]: I guess these comments could then be removed

File: .travis.yml [hunk 2]
```diff
@@ -71,13 +74,15 @@ jobs:
...
     - stage: test
+      name: 'Win64'
       env: >-
         HOST=x86_64-w64-mingw32
         PACKAGES="python3 nsis g++-mingw-w64-x86-64 wine-binfmt wine64"
         GOAL="deploy"
         BITCOIN_CONFIG="--enable-reduce-exports --disable-gui-tests"
 # 32-bit + dash
     - stage: test
+      name: '32-bit + dash'
```
Comment (scravy) [inline .travis.yml]: I for one value very much to be able to see the build environment variables when looking at the travis overview page for the build. This is lost now as one sees only this identifier. I like to see the environment variables to understand whether the particular build runs functional tests and/or unit tests or not. Maybe this information should be added to the name.

**Comment by scravy** [comment]
Documentation for the feature: https://docs.travis-ci.com/user/build-stages/#naming-your-jobs-within-build-stages

File: .travis.yml (hunk 2 above)
Comment (gkrizek) [reply .travis.yml]: I agree! I was sad to see that Travis removed the environment variables from the UI when you add a name. You can see the environment variables still if you go into a build job. Nonetheless, I can certainly add a test type to the job name if we would like.

File: .travis.yml (hunk 1 above)
Comment (gkrizek) [reply .travis.yml]: I just pushed a commit that removes these comments and moves the needed comments to beside the job name.

**Comment by Empact** [comment]
Concept ACK

**Comment by gkrizek** [comment]
As previously commented, when you add a name to your jobs the Travis UI removes the environment variables from the build when in the list view. (They still show up when you view an individual build's page). I think having a build name is more valuable in the list view than the variables. We can always add some sort of 'tag' to the job name if we feel that's needed.

**Comment by fanquake** [comment]
Here's the current Travis setup vs this PR:
Current:
current
#15020 (30922a7):
build names

If anything, I think we could add more build related information to the "name", if we go down that route.

**Comment by Empact** [comment]
Here's an example where a new "disable bip 70" build was added - I challenge you to quickly identify which build it is: https://travis-ci.org/bitcoin/bitcoin/builds/473735618?utm_source=github_status&utm_medium=notification

It could be helpful to describe the build more fully, e.g. "no depends, only system libs, sanitizers: thread (TSan)" is relevant IMO. Also possibly: `GOAL`, osx version.

**Comment by gkrizek** [comment]
@Empact That's why I thought this PR could be helpful. Same for someone new looking at the project. Very hard to tell what's going on in the current state.

I just pushed a commit that adds more details to the job names as it seems like more are in agreement that it would be helpful.

File: .travis.yml [hunk 3]
```diff
@@ -61,65 +63,74 @@ jobs:
         # -Wno-psabi is to disable ABI warnings: "note: parameter passing for argument of type ... changed in GCC 7.1"
         # This could be removed once the ABI change warning does not show up by default
         BITCOIN_CONFIG="--enable-glibc-back-compat --enable-reduce-exports CXXFLAGS=-Wno-psabi"
-# Win32
+
     - stage: test
+      name: 'Win32  [GOAL: deploy]'
```
Comment (fanquake) [inline .travis.yml]: Could add `[no gui tests]` to Win32 & Wind64.

File: .travis.yml [hunk 4]
```diff
@@ -61,65 +63,74 @@ jobs:
...
       env: >-
         HOST=x86_64-unknown-linux-gnu
         PACKAGES="python3-zmq qtbase5-dev qttools5-dev-tools protobuf-compiler libdbus-1-dev libharfbuzz-dev libprotobuf-dev"
         DEP_OPTS="NO_QT=1 NO_UPNP=1 DEBUG=1 ALLOW_HOST_PACKAGES=1"
         GOAL="install"
         BITCOIN_CONFIG="--enable-zmq --with-gui=qt5 --enable-glibc-back-compat --enable-reduce-exports --enable-debug CXXFLAGS=\"-g0 -O2\""
-# x86_64 Linux (xenial, no depends, only system libs, sanitizers: thread (TSan))
+
     - stage: test
+      name: 'x86_64 Linux  [GOAL: install]  [xenial]  [no depends, only system libs, sanitizers: thread (TSan)]'
```
Comment (fanquake) [inline .travis.yml]: Can add `[no wallet]` to the last set of brackets here. See `--disable-wallet`.

File: .travis.yml [hunk 5]
```diff
@@ -61,65 +63,74 @@ jobs:
...
       env: >-
         HOST=x86_64-unknown-linux-gnu
         PACKAGES="python3-zmq"
         DEP_OPTS="NO_WALLET=1"
         GOAL="install"
         BITCOIN_CONFIG="--enable-glibc-back-compat --enable-reduce-exports"
-# Cross-Mac
+
     - stage: test
+      name: 'macOS 10.11  [GOAL: deploy]'
```
Comment (fanquake) [inline .travis.yml]: This should be `macOS 10.10`.

File: .travis.yml [hunk 6]
```diff
@@ -61,65 +63,74 @@ jobs:
...
         DOCKER_NAME_TAG=ubuntu:16.04
         PACKAGES="clang llvm python3-zmq qtbase5-dev qttools5-dev-tools libssl-dev libevent-dev bsdmainutils libboost-system-dev libboost-filesystem-dev libboost-chrono-dev libboost-test-dev libboost-thread-dev libdb5.3++-dev libminiupnpc-dev libzmq3-dev libprotobuf-dev protobuf-compiler libqrencode-dev"
         NO_DEPENDS=1
+        FUNCTIONAL_TESTS_CONFIG="--exclude feature_block.py,p2p_invalid_messages.py"
         GOAL="install"
         BITCOIN_CONFIG="--enable-zmq --disable-wallet --with-gui=qt5 CPPFLAGS=-DDEBUG_LOCKORDER --with-sanitizers=thread --disable-hardening --disable-asm CC=clang CXX=clang++"
-# x86_64 Linux (no depends, only system libs, sanitizers: address/leak (ASan + LSan) + undefined (UBSan) + integer)
+
     - stage: test
+      name: 'x86_64 Linux  [GOAL: install]  [no depends, only system libs, sanitizers: address/leak (ASan + LSan) + undefined (UBSan) + integer]'
```
Comment (fanquake) [inline .travis.yml]: Could add `[bionic]` here and to the other Linux builds to make it explicit/obvious which OS is being used.

**Comment by gkrizek** [comment]
@fanquake Thanks for the review! I've pushed a commit with your changes added.

**Comment by fanquake** [comment]
ACK 5af7d0b. I think this is ok now.

travis named jobs

@scravy @Empact thoughts?

File: .travis.yml [hunk 7]
```diff
@@ -61,65 +63,74 @@ jobs:
...
-# x86_64 Linux (xenial, no depends, only system libs, sanitizers: thread (TSan))
+
     - stage: test
+      name: 'x86_64 Linux  [GOAL: install]  [xenial]  [no depends, only system libs, sanitizers: thread (TSan), no wallet]'
       env: >-
         HOST=x86_64-unknown-linux-gnu
         DOCKER_NAME_TAG=ubuntu:16.04
         PACKAGES="clang llvm python3-zmq qtbase5-dev qttools5-dev-tools libssl-dev libevent-dev bsdmainutils libboost-system-dev libboost-filesystem-dev libboost-chrono-dev libboost-test-dev libboost-thread-dev libdb5.3++-dev libminiupnpc-dev libzmq3-dev libprotobuf-dev protobuf-compiler libqrencode-dev"
         NO_DEPENDS=1
+        FUNCTIONAL_TESTS_CONFIG="--exclude feature_block.py,p2p_invalid_messages.py"
```
Comment (MarcoFalke) [inline .travis.yml]: Why?

**Comment by laanwj** [comment]
Concept ACK

File: .travis.yml (hunk 7 above)
Comment (laanwj) [reply .travis.yml]: looks like an accidental revert of fa05d52af475dc4109c09b70b17b22f5ffc6e6e6

Comment (gkrizek) [reply .travis.yml]: Ah, I think you are right. I'll fix that

Comment (gkrizek) [reply .travis.yml]: @MarcoFalke @laanwj  I just pushed a new commit without this line. Sorry about that.

**Comment by jnewbery** [comment]
Great! I've been wanting this for a long time. utACK 1db71d4a29356b7c3e3ea3c806cff586148dcb65.

**Pull request merged**